OpenSim/Simulation/Model/Frame.cpp

```diff
@@ -128,4 +128,16 @@ void Frame::extendRealizeTopology(SimTK::State& s) const
 {
     Super::extendRealizeTopology(s);
     groundTransformIndex = getCacheVariableIndex("ground_transform");
+}
+void Frame::extendRealizeVelocity(const SimTK::State& s) const
+{
+  Super::extendRealizeVelocity(s);
+
+  SimTK::Value<SimTK::Transform>::downcast(
+      getSystem().getDefaultSubsystem().
+      updCacheEntry(s, groundTransformIndex)).upd()
+          = calcGroundTransform(s);
+  // mark cache as up-to-date
+  getSystem().getDefaultSubsystem().
+      markCacheValueRealized(s, groundTransformIndex);
```

Comment:
The indent here should be 4 spaces.

I think this updating of the cache entry still only needs to happen if it is not vaild.